modules/create-a-containerruntimeconfig-crd.adoc

```diff
@@ -0,0 +1,125 @@
+The ContainerRuntimeConfigs custom resource definition (CRD) provides a
+structured way of changing settings associated with the {product-title}
+CRI-O runtime. Using a ContainerRuntimeConfigs custom resource (CR),
+you select the configuration values you want and the MCO handles rebuilding
+the `crio.conf` and `storage.conf` configuration files.
+
+Parameters you can set in a ContainerRuntimeConfig CR include:
+
+* **PIDs limit**: Sets the maximum number of processes allowed in a container.
+By default, the limit is set to 1024 (`pids_limit = 1024`).
+* **Log level**: Sets the level of verbosity for log messages. The default is
+`info` (`log_level = info`). Other options include `fatal`, `panic`, `error`,
+`warn`, `debug`, and `trace`.
+* **Overlay size**: Sets the maxim size of a container image. The default is
+10GB.
+* **Maximum log size**: Sets the maximum size allowed for the container
+log file. The default maximum log size is unlimited (`log_size_max = -1`).
+If it is set to a positive number, it must be at least 8192
+to not be smaller than conmon's read buffer.
+
+The following procedure illustrates how to change CRI-O settings using the
+ContainerRuntimeConfig CR.
+
+.Procedure
+
+. To raise the pidsLimit to 2048, set the logLevel to debug, and
+set the overlaySize to 8GB, create a custom resource file
+(for example, `overlay-size.yaml`) that contains that setting:
++
+[source,yaml]
+----
+$ cat << EOF > /tmp/overlay-size.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+ name: overlay-size
+spec:
+ machineConfigPoolSelector:
+   matchLabels:
+     custom-crio: overlay-size
+ containerRuntimeConfig:
+   pidsLimit: 2048
+   logLevel: debug
+   overlaySize: 8G
+EOF
+----
+
+. To apply the ContainerRuntimeConfig settings, run:
++
+[source,terminal]
+----
+$ oc create -f /tmp/overlay-size
+----
+
+.. Run:
++
+[source,terminal]
+----
+$ oc get ContainerRuntimeConfig
+NAME           AGE
+overlay-size   3m19s
+
+----
+
+. Open a MachineConfigPool for editing for a pool of machines (in this case, `worker`).
++
+[source,terminal]
+----
+$ oc edit machineconfigpool worker
+----
+
+. Modify the MachineConfigPool to add the new ContainerRuntimeConfig CR.
+In this case, add `custom-crio: overlay-size` under `labels`:
++
+[source,terminal]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  creationTimestamp: "2020-09-27T14:45:10Z"
+  generation: 11
+  labels:
+    custom-crio: overlay-size
+    machineconfiguration.openshift.io/mco-built-in: ""
+----
+
+. Check that a new containerruntime object has appeared under the machineconfigs:
++
+[source,terminal]
+----
+$ oc get machineconfigs | grep containerrun
+99-worker-generated-containerruntime   2c9371fbb673b97a6fe8b1c52691999ed3a1bfc2  3.1.0  31s
+----
+. Monitor the machineconfigpool as the changes are rolled into the machines until all are shown as ready:
```

Comment:
s/machineconfigpool/MachineConfigPool/ ?
(or "machine config pool" if this is the generic concept)